src/Benraz.Infrastructure.Files/GCP/GcpBucketFilesService.cs

```diff
@@ -0,0 +1,331 @@
+    public async Task<File> FindByNameAsync(string fileName, FileProperties fileProperties)
+    {
+        try
+        {
+            var obj = await _storageClient.GetObjectAsync(_settings.BucketName, fileName);
+
+            var file = new File();
+
+            if (fileProperties.HasFlag(FileProperties.Metadata) || fileProperties.HasFlag(FileProperties.All))
+            {
+                file.Name = obj.Name;
+                file.Uri = GetUri(obj.Name);
+                file.CreateTimeUtc = obj.TimeCreated?.ToUniversalTime();
+                file.UpdateTimeUtc = obj.Updated?.ToUniversalTime();
+            }
+
+            if (fileProperties.HasFlag(FileProperties.Content) || fileProperties.HasFlag(FileProperties.All))
+            {
+                using var memoryStream = new MemoryStream();
+                await _storageClient.DownloadObjectAsync(obj, memoryStream);
+                file.Content = memoryStream.ToArray();
+            }
+
+            return file;
+        }
+        catch (GoogleApiException ex) when (ex.HttpStatusCode == System.Net.HttpStatusCode.NotFound)
+        {
+            return null;
+        }
+
+    }
+
+    /// <summary>
+    /// Creates file in bucket and returns file with URI updated.
+    /// </summary>
+    /// <param name="file">File.</param>
+    /// <returns>File.</returns>
+    public async Task<File> CreateAsync(File file)
+    {
+        using var stream = new MemoryStream(file.Content);
+        var obj = await _storageClient.UploadObjectAsync(
+            bucket: _settings.BucketName,
+            objectName: file.Name,
+            contentType: GetContentType(file.Name),
+            source: stream
+        );
+
+        file.Uri = GetUri(file.Name);
+        file.CreateTimeUtc = obj.TimeCreated?.ToUniversalTime();
+        file.UpdateTimeUtc = obj.Updated?.ToUniversalTime();
+
+        return file;
+    }
+
+    /// <summary>
+    /// Renames file and returns file with name and URL updated.
+    /// </summary>
+    /// <param name="file">File to rename.</param>
+    /// <param name="newFileName">New file name.</param>
+    /// <returns>Updated file.</returns>
+    public async Task<File> RenameAsync(File file, string newFileName)
+    {
+        var destinationObject = await _storageClient.CopyObjectAsync(
+            _settings.BucketName, file.Name, _settings.BucketName, newFileName
+        );
+
+        await _storageClient.DeleteObjectAsync(_settings.BucketName, file.Name);
+
+        file.Name = newFileName;
+        file.Uri = GetUri(newFileName);
+        file.UpdateTimeUtc = destinationObject.Updated?.ToUniversalTime();
+
+        return file;
+    }
+
+    /// <summary>
+    /// Deletes file with specified name from container.
+    /// </summary>
+    /// <param name="fileName">File name.</param>
+    /// <returns>Task.</returns>
+    public async Task DeleteAsync(string fileName)
+    {
+        await _storageClient.DeleteObjectAsync(_settings.BucketName, fileName);
```

Comment:
@parimal1987 @fuadagazade why no indication if passed or any issue? shouldnt return status?
if this how we implement Blob and Ftp then leave it, then just for the record